scripts/retry-helm.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+
+# Runs the helm CLU with supplied arguments and retries on failures indicating network errors.
+#
+# To detect such transient errors, stderr from helm is redirected to a file and grepped for certain patterns
+# in case helm exits with an error code.
+#
+# To make helm invocations a little bit more idempotent in the presence of failures we are injecting
+# the --atomic flag in case the helm command is `install` or `upgrade`.
+
+set -euo pipefail
+
+# RegExes for which we attempt a retry (one regex per line).
+error_regex=$(tr '\n' '|' <<EOT | sed -e 's/|$//;'
+: the server is currently unable to handle the request
+EOT
+)
+
+helm_atomic_wait_timeout=10m
+
+tmp_in="$(mktemp)"
+tmp_out="$(mktemp --suffix=-stdout.txt)"
+tmp_err="$(mktemp --suffix=-stderr.txt)"
+grep_out="$(mktemp)"
+trap 'cat ${tmp_out}; cat ${tmp_err} >&2; rm -f ${tmp_in} ${tmp_out} ${tmp_err} ${grep_out}' EXIT
+
+inject_atomic_flag() {
```

Comment:
This function is quite scary with its array-by-reference passing, and the index-based manipulation.
Would it be possible to instead:
1. make it just either set `extra_args` to the list of two arguments or an empty list.
2. in the main loop just call `"${HELM_CMD:-helm}" "$@" "${extra_args[@]}" > "${tmp_out}" 2>"${tmp_err}"`?